Add MFEM mesh refinement to steady state problems (#30943)
idaholab/moose: framework/doc/content/source/mfem/markers/MFEMRefinementMarker.md

# MFEMRefinementMarker

!if! function=hasCapability('mfem')

## Summary

Wrapper around `mfem::ThresholdRefiner`. Input file needs to have the name of an estimator to construct correctly.

> 

Reviewer: ```suggestion
```

## Overview

Currently contains one method for h-refinement (`HRefine`) and another for p-refinement (`MarkWithoutRefining`).
Both advance an internal counter which determines how many refinement steps have been taken.

> 

Reviewer: ```suggestion
## Example Input File Syntax

!listing mfem/kernels/diffusion_amr.i block=Adaptivity

!syntax parameters /Adaptivity/Markers/MFEMRefinementMarker

!syntax inputs /Adaptivity/Markers/MFEMRefinementMarker

!syntax children /Adaptivity/Markers/MFEMRefinementMarker
```

# MFEMRefinementMarker

!if! function=hasCapability('mfem')

## Summary

Wrapper around `mfem::ThresholdRefiner`. Input file needs to have the name of an estimator to construct correctly.

## Overview

> Currently contains one method for h-refinement (`HRefine`) and another for p-refinement (`MarkWithoutRefining`).

Reviewer: ```suggestion
Currently contains one method for h-refinement (`hRefine`), which refines the mesh, and another for p-refinement (`pRefineMarker`), which marks what elements to refine.
```
